src/typeahead/typeahead-window.ts

```diff
@@ -73,17 +84,60 @@ export class NgbTypeaheadWindow implements OnInit {
   @Input() resultTemplate: TemplateRef<ResultTemplateContext>;
 
   /**
+   * A CSS max-height value to limit the height and be able to scroll through the list of results.
+   */
+  @Input() maxHeight: string;
+
+  /**
    * Event raised when user selects a particular result row
    */
   @Output('select') selectEvent = new EventEmitter();
 
   @Output('activeChange') activeChangeEvent = new EventEmitter();
 
+  constructor(private view: ViewContainerRef) {}
+
+  ngOnDestroy() {
+    if (this._mouseMoveListener != null) {
+      document.removeEventListener('mousemove', this._mouseMoveListener);
+    }
+  }
+
   getActive() { return this.results[this.activeIdx]; }
 
+  onMouseLeaveDropdown() {
+    // we don't want to handle the mouse anymore as soon as it is outside of
+    // the dropdown
+    this._latestMousePosition = null;
+    if (this._mouseMoveListener != null) {
+      document.removeEventListener('mousemove', this._mouseMoveListener);
+      this._mouseMoveListener = null;
+    }
+  }
+
+  onMouseEnterItem(event, activeIdx) {
+    // to prevent the case where the mouse cursor stays over the dropdown
+    // while we scroll the content because of a keyboard navigation
+    // scrolling makes a the mouse enter a new element,
+    // and therefore would mark it active otherwise
+    const latestMousePosition = this._latestMousePosition;
+    const x = event.screenX;
+    const y = event.screenY;
+
+    if (latestMousePosition == null || latestMousePosition.x !== x || latestMousePosition.y !== y) {
+      this.markActive(activeIdx);
+    }
+
+    // to keep track of the actual latest mouse position while we keep the mouse inside items
+    if (this._mouseMoveListener == null) {
+      this._mouseMoveListener = (moveEvent) => this._latestMousePosition = {x: moveEvent.screenX, y: moveEvent.screenY};
+      document.addEventListener('mousemove', this._mouseMoveListener);
```

Comment:
- This will trigger a huge number of change detections. Just add `ngDoCheck() {}` to see it
- You should use `renderer.listen(...)` to register listeners